### PR TITLE
Implement the built-in logging provider

### PR DESCRIPTION
**Pull request by autodidaddict**
Signed-off-by: Kevin Hoffman <autodidaddict@users.noreply.github.com>

### CODE REVIEW

**Comment by autodidaddict** [comment]
@brooksmtownsend looking for some feedback on the invocation refactor before I move on to adding in support for the built-in logging provider. We're going to repeat this pattern for other built-in providers so I want to get it clean first.

File: host_core/lib/host_core/web_assembly/imports.ex [hunk 1]
```diff
@@ -121,132 +121,148 @@ defmodule HostCore.WebAssembly.Imports do
...
     0
   end
 
-  defp finish_host_call(
-         {:ok, actor, binding, namespace, operation, payload, _claims, seed, prefix, provider_key,
-          state},
-         agent
+  defp invoke(
+         _token = %{
+           authorized: true,
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/web_assembly/imports.ex:220]: I struggled with this when refactoring, is there any variation of function names that makes `invoke` and `do_invoke` less confusing? I like this better than `finish_host_call` and `invoke`, 🤔 

Thoughts on the top-level pipeline to be:
```
token
    |> identify_target()
    |> verify_link()
    |> authorize_call()
    |> generate_invocation()
    |> invoke()
```

Just a brainstorm. If this feels worse to you then I'm happy with where it is

Comment (autodidaddict) [reply host_core/lib/host_core/web_assembly/imports.ex:239]: refactored the names here

**Comment by autodidaddict** [comment]
@brooksmtownsend pushed the initial implementation of the builtin logging provider. I'm unsure if there's a good testing strategy here until we've got some new actors that actually use logging. Even then, not sure we'll be able to do anything like reliably capture stdout to ensure that logging is happening other than just eyeballing it when we run a manual test.

File: host_core/lib/host_core/web_assembly/imports.ex [hunk 2]
```diff
@@ -2,6 +2,8 @@ defmodule HostCore.WebAssembly.Imports do
   require Logger
   alias HostCore.Actors.ActorModule.State
 
+  @wasmcloud_logging "wasmcloud:builtin:logging"
+
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/web_assembly/imports.ex:6]: we already mentioned it in slack but we just have to remember to change wash and tell others to re-sign

File: host_core/lib/host_core/web_assembly/imports.ex [hunk 3]
```diff
@@ -121,132 +123,166 @@ defmodule HostCore.WebAssembly.Imports do
...
+        seed,
+        actor,
+        target_type,
+        target_key,
+        namespace,
+        binding,
+        operation,
+        payload
+      )
+      |> perform_rpc_invoke(target_subject)
```
Comment (brooksmtownsend) [inline host_core/lib/host_core/web_assembly/imports.ex:262]: I like this 👍🏻

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**